WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -4,7 +4,7 @@ class RegistrationsController < ApplicationController
 
   private def competition_from_params
     competition = if params[:competition_id]
-                    Competition.includes(:events).find(params[:competition_id])
+                    Competition.find(params[:competition_id])
```

Comment:
The include is probably faster with a multi competition query.

Event constants in memory would be even faster :0